compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -845,19 +864,6 @@ class ExpressionCodegen(
         }
     }
 
-    private fun IrExpression.markEndOfStatementIfNeeded() {
```

Comment:
These generate line numbers for the closing brace of `when`, `try`, `if`. However, this does not ensure that there is an instruction for that line number. Therefore, these line numbers will be there in the line number table, but will often not have instructions and can therefore not be observed. With the old backend, these are also emitted and typically cannot be observed. I opted for removing them entirely as I do not find it useful to be able to step and set breakpoints on the closing brace. The alternative is to go the other way and always emit a `nop` so that you will always get a step here (and the posibility of setting a breakpoint).